### PR TITLE
Add Doc Sections about Credentials and Authentication

### LINKED ISSUE #116: Documentation: GCP credentials
Default app credentials vs service accounts.

### PR DESCRIPTION
**Pull request by dzou**
Adds sections about authentication and credentials to the docs.

Fixes #116.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,14 @@ An implementation of the [R2DBC](https://r2dbc.io/) driver for [Cloud Spanner](h
 
 ## Setup Instructions
 
-This section describes how to setup and begin using the Cloud Spanner R2DBC driver.
-Below are the dependencies to add to your build configuration.
+The sections below describe how to setup and begin using the Cloud Spanner R2DBC driver.
+
+An overview of the steps are as follows:
```
Comment (ChengyuanZhao) [inline README.md]:
```suggestion
An overview of the steps is as follows:
```

File: README.md [hunk 2]
```diff
@@ -6,8 +6,14 @@ An implementation of the [R2DBC](https://r2dbc.io/) driver for [Cloud Spanner](h
...
+
+1. Add the Cloud Spanner R2DBC driver dependency to your build configuration.
+2. Configure the driver credentials/authentication for your Google Cloud Platform project to access
+    Cloud Spanner.
+3. Instantiate the R2DBC `ConnectionFactory` in Java code to build Connections and run queries.
```
Comment (ChengyuanZhao) [inline README.md]: Perhaps a code snippet about making this connection factory? Its constructor isn't user-friendly :/

File: README.md (hunk 1 above)
Comment (dzou) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (dzou) [reply README.md]: Yes, this is provided in the Usage section. The few sentences here were intended to be an outline of this setup section with elaboration provided below.

File: README.md [hunk 3]
```diff
@@ -19,16 +31,61 @@ Below are the dependencies to add to your build configuration.
...
+
+#### Service Account
+
+A [Google Service Account](https://cloud.google.com/iam/docs/understanding-service-accounts) is a
+special type of Google Account intended to represent a non-human user that needs to authenticate
+and be authorized to access your Google Cloud resources. Each service account has an account key
+JSON file that is used as the credentials to accessing the resources of your account.
+This is the recommended method of authentication for production use.
+
+Setting up credentials for your service account can be done by following [these instructions](https://cloud.google.com/iam/docs/creating-managing-service-accounts).
```
Comment (meltsufin) [inline README.md]: Mention the `google_credentials` option somewhere here?

Comment (dzou) [reply README.md]: Done.

File: README.md [hunk 4]
```diff
@@ -19,16 +31,81 @@ Below are the dependencies to add to your build configuration.
...
+
+You can learn how to create a service account and authenticate your application by following
+[these instructions](https://cloud.google.com/docs/authentication/production#obtaining_and_providing_service_account_credentials_manually).
+
+If you are unsatisfied with credentials inference methods, you may override this behavior
+by manually specifying a service account key JSON file using the `google_credentials`
+option to the `ConnectionFactory` builder.
+
+Example:
+```
```
Comment (meltsufin) [inline README.md]:
```suggestion
```java
```

File: README.md [hunk 5]
```diff
@@ -19,16 +31,81 @@ Below are the dependencies to add to your build configuration.
...
+### Authentication
+
+By default, the R2DBC driver will attempt to infer your account credentials from the environment
+in which the application is run. There are a number of different ways to conveniently provide
+account credentials to the driver.
+
+#### Using Google Cloud SDK
+
+Google Cloud SDK is a command line interface for Google Cloud Platform products and services.
+This is the recommended way of setting up authentication during local development.
```
Comment (elefeint) [inline README.md]: I think service accounts are recommended; gcloud auth is just convenient.

Comment (dzou) [reply README.md]: Done; updated 'recommended' with 'convenient'. Emphasized sentence that service accounts is recommended below.

**Pull request merged**